generator/prompts.ts

```diff
@@ -19,29 +19,39 @@ function isAddressOrKeepCurrent(value: string) {
 }
 
 // TRANSFORMS
-function transformNumberToPercent(value: string) {
+export function transformNumberToPercent(value: string) {
   if (value && isNumber(value)) {
-    if (Number(value) <= 9) value = value.padStart(2, '0');
-    return value.replace(/(?=(\d{2}$)+(?!\d))/g, '.') + ' %';
+    return (
+      new Intl.NumberFormat('en-us', {
+        maximumFractionDigits: 2,
+      }).format(value as unknown as number) + ' %'
+    );
   }
   return value;
 }
 
-function transformNumberToHumanReadable(value: string) {
+export function transformNumberToHumanReadable(value: string) {
   if (value && isNumber(value)) {
-    return value.replace(/(?=(\d{3}$)+(?!\d))/g, '.');
+    return new Intl.NumberFormat('en-us').format(BigInt(value));
```

Comment:
When the value is a decimal (inputted via the keyboard) and when trying to convert it into `BigInt()` it throws an error and reverts. Maybe a good idea to validate that the number is not a decimal before and handle it gracefully.

Reply:
@brotherlymite fair point.

Instead of doing this though i went ahead and created a new input prompt that allows specifying a pattern that must be followed (so numberInput will only allow numbers, percentInput only allows a single decimal point,... more to come i guess)

Reply:
@brotherlymite fair point.

Instead of doing this though i went ahead and created a new input prompt that allows specifying a pattern that must be followed (so numberInput will only allow numbers, percentInput only allows a single decimal point,... more to come i guess)

Reply:
Thx, looks good now